cwms_radar_api/src/main/java/cwms/radar/api/TimeSeriesController.java

```diff
@@ -126,7 +207,7 @@ public void getAll(Context ctx) {
                 final Timer.Context timeContext = getAllRequestsTime.time();
                 DSLContext dsl = getDslContext(ctx))
         {
-            TimeSeriesDao dao = new TimeSeriesDao(dsl);
+            TimeSeriesDao dao = getTimeSeriesDao(dsl);
```

Comment:
we don't need to do this "right now" but would it make more sense to have the DAOs implement AutoClosable and put them up in the try-with-resources block?

Reply:
I'm torn about how the *Controller and *Dao and DSLContext and Connection should work together.  Before Javalin I was coming from cwms mobile and Spring.   In Spring the Controllers, Dao and Services were long-lived objects and used @autowired DataSource or JdbcTemplate to build new Connections as needed.  The Javalin Dao are structured the way it currently is b/c I wanted to see if we could make all the Dao that used Jooq not have to know anything about Connection or Context.   But I hadn't done much development around jOOQ or DSLContext.  Its easier to get a Connection from DSLContext than I first realized and it takes less to construct a DSLContext than I initially realized.  I also had this idea that we'd be able to swap in mocks for the DSLContext and we'd get this cool powerful way to do testing.  But when I tried the DSLContext mocking it didn't work anywhere near the way I wanted it to.  My inclination now would be to have the JooqDao objects not have a private DSLContext field and instead to just have every public dao method take a Connection as one of the arguments.  Then it wouldn't matter if the Dao were short-lived or long-lived.